### PR TITLE
Fix part of #210: Add tests for TextInput's contains rule classifier

### PR DESCRIPTION
**Pull request by miaboloix**
## Explanation

Fixes part of #210
Added tests for TextInput's contains rule classifier

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    val inputs = mapOf("x" to STRING_ANSWER)
+
+    val matches = inputContainsRuleClassifier.matches(answer = STRING_ANSWER, inputs = inputs)
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testStringAnswer_stringInput_answerContainsInputAtBeginning_bothValuesMatch() {
+    val inputs = mapOf("x" to STRING_INPUT_BEGINNING)
```
Comment (BenHenning) [inline domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt]: Hmm, since contains is heavily nuanced on the exact value of the inputs it may be worth just inlining all of the constants for these tests. Without them, it's hard to verify the test is testing what it's supposed to be without looking outside of the test body.

File: domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [TextInputContainsRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class TextInputContainsRuleClassifierProviderTest {
```
Comment (BenHenning) [inline domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt:24]: Is it worth also adding tests for an empty input string, and an empty answer string?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @miaboloix! One thing I forgot to mention: here and in your other PRs, can you remove the corresponding TODO on issue #210 to add tests in the provider class for which you're adding tests?

File: domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt (hunk 2 above)
Comment (miaboloix) [reply domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt:24]: For sure! Will add that now, thanks!

File: domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt (hunk 1 above)
Comment (miaboloix) [reply domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt]: OK, will do!

Comment (miaboloix) [reply domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt]: Should I inline the constant I named "STRING_ANSWER"? I see why it could be confusing and require those without context to look outside the test to understand what is going on, but at the same time I use that string in almost every test - isn't it good practice to have it be a constant when you use it so much? And if I do keep it a constant, maybe re-naming it to something like "STRING_THIS_IS_A_TEST" or something like that will help?

**Review by BenHenning — APPROVED** [review]
Thanks @miaboloix -- this LGTM!
